pennylane/templates/subroutines/grover.py

```diff
@@ -113,7 +113,10 @@ def __init__(self, wires=None, work_wires=None, id=None):
         if (not hasattr(wires, "__len__")) or (len(wires) < 2):
             raise ValueError("GroverOperator must have at least two wires provided.")
 
-        self._hyperparameters = {"n_wires": len(wires), "work_wires": Wires(work_wires)}
+        self._hyperparameters = {
+            "n_wires": len(wires),
+            "work_wires": Wires(work_wires) if work_wires else Wires([]),
```

Comment:
in case someone sets `0` as the work wires.
```suggestion
            "work_wires": Wires(work_wires) if work_wires is not None else Wires([]),
```